test/curry.js

```diff
@@ -76,6 +76,36 @@ describe('curry', function() {
     assert.deepEqual(g(_, _, _)(1, _, _)(_, _)(2, _)(_)(3), [1, 2, 3]);
   });
 
+  it('supports @@functional/placeholder', function() {
+    var f = function(a, b, c) { return [a, b, c]; };
+    var g = R.curry(f);
+    var _ = {'@@functional/placeholder': true, x: Math.random()};
+
+    assert.deepEqual(g(1)(2)(3), [1, 2, 3]);
+    assert.deepEqual(g(1)(2, 3), [1, 2, 3]);
+    assert.deepEqual(g(1, 2)(3), [1, 2, 3]);
+    assert.deepEqual(g(1, 2, 3), [1, 2, 3]);
+
+    assert.deepEqual(g(_, 2, 3)(1), [1, 2, 3]);
+    assert.deepEqual(g(1, _, 3)(2), [1, 2, 3]);
+    assert.deepEqual(g(1, 2, _)(3), [1, 2, 3]);
+
+    assert.deepEqual(g(1, _, _)(2)(3), [1, 2, 3]);
+    assert.deepEqual(g(_, 2, _)(1)(3), [1, 2, 3]);
+    assert.deepEqual(g(_, _, 3)(1)(2), [1, 2, 3]);
+
+    assert.deepEqual(g(1, _, _)(2, 3), [1, 2, 3]);
+    assert.deepEqual(g(_, 2, _)(1, 3), [1, 2, 3]);
+    assert.deepEqual(g(_, _, 3)(1, 2), [1, 2, 3]);
+
+    assert.deepEqual(g(1, _, _)(_, 3)(2), [1, 2, 3]);
+    assert.deepEqual(g(_, 2, _)(_, 3)(1), [1, 2, 3]);
+    assert.deepEqual(g(_, _, 3)(_, 2)(1), [1, 2, 3]);
+
+    assert.deepEqual(g(_, _, _)(_, _)(_)(1, 2, 3), [1, 2, 3]);
+    assert.deepEqual(g(_, _, _)(1, _, _)(_, _)(2, _)(_)(3), [1, 2, 3]);
+  });
+
```

Comment:
Well, since `curry` is public and its behavior has changed, it's tests should probably reflect that.  The fact that it is implemented in terms of `curryN` is not a public guarantee, so our tests shouldn't assume this.

OTOH, we probably also need similar tests for `curryN`, as that one is also public.